angular-client/src/app/app.module.ts

```diff
@@ -190,19 +191,21 @@ import { BatteryInfoDisplay } from 'src/pages/charging-page/components/battery-i
     DynamicDialogModule,
     BrowserAnimationsModule,
     ButtonModule,
-    HttpClientModule,
     MatIconModule,
     SidebarModule,
     MatToolbarModule,
     MatButtonModule,
     MatInputModule,
     ReactiveFormsModule
   ],
-  providers: [DialogService, MessageService],
-  bootstrap: [AppContext]
+  providers: [
+    DialogService,
+    MessageService,
+    provideHttpClient(withInterceptorsFromDi())
```

Comment:
Angular is going "standalone". The `HttpClientModule` is deprecated and replaced by the standalone provider factory `provideHttpClient`

Reply:
The new http interceptor api uses functional interceptors. In order to enable class-based interceptors, `withInterceptorsFromDi` is necessary.